internal/config/options.go

```diff
@@ -148,6 +155,125 @@ func NewOptions() *Options {
 	return o
 }
 
+// OptionsFromViper builds the main binary's configuration
+// options by parsing environmental variables and config file
+func OptionsFromViper(configFile string) (*Options, error) {
+	o := NewOptions()
+
+	// Load up config
+	o.bindEnvs()
+	if configFile != "" {
+		log.Info().Msgf("Loading config from '%s'", configFile)
+		fh, err := os.Open(configFile)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to open config file '%s': %s", configFile, err)
+		}
+		err = viper.ReadConfig(fh)
+		if err != nil {
+			log.Fatal().Msgf("Failed to read config: %s", err)
```

Comment:
An error is returned if the file cannot be opened but panics if it can't be read by viper. 
Curious, why throw a panic here vs an error above; can we just return an error? 
So for example, I can imagine a scenario where a policy file is updated and for-whatever reason the config file can't be read, I'd rather pomerium keep running with the outdated policy than panic (and recover?).

Reply:
Yes; agreed.  This should be consistent.  Main can figure out what to do.